release/Dockerfile.tiny

```diff
@@ -16,8 +19,9 @@ RUN rm -Rf /pudl/repo
 # because it is needed for the pip install.
 
 FROM python:3.8
-# TODO(rousik): use python:3.8-slim or python:3.8-slim-buster
 RUN apt-get update && apt-get -y install libsnappy-dev git
+COPY --from=pudl-source-build /pudl/package/requirements.txt /tmp/requirements.txt
+RUN --mount=type=cache,mode=0755,target=/root/.cache/pip pip install -r /tmp/requirements.txt
```

Comment:
Curious to understand what all is going on here. When things are working can you give us a tour through how all these parts are working together?